pkg/dns/dns.go

```diff
@@ -166,11 +166,16 @@ func (kd *KubeDNS) waitForResourceSyncedOrDie() {
 		case <-timeout:
 			glog.Fatalf("Timeout waiting for initialization")
 		case <-ticker.C:
-			if kd.endpointsController.HasSynced() && kd.serviceController.HasSynced() {
-				glog.V(0).Infof("Initialized services and endpoints from apiserver")
-				return
+			if !kd.endpointsController.HasSynced() {
+				glog.V(0).Infof("Waiting for endpoints to be initialized from apiserver...")
+				continue
+			}
+			if !kd.serviceController.HasSynced() {
+				glog.V(0).Infof("Waiting for services to be initialized from apiserver...")
```

Comment:
It is also confusing if both endpoints and services fail to be synced while this only says waiting for endpoint. Could you make it check both `endpointsController` and `serviceController` regardless?

Reply:
@MrHohn Thanks for your review, I got it.
I will update this PR soon based on your comment.